FreeRTOS/Demo/ARC_NSIM_QEMU_GCC_MW/Readme.md

```diff
@@ -0,0 +1,78 @@
+FreeRTOS Demo
+====================
+
+Overview
+====================
+ This example demostrates basic FreeRTOS task usage.
+
+Requirements
+====================
+ * Toolchains:
+
+    Metaware or arc_gnu
+
+ * Simulators:
+
+    nSIM or QEMU
+
+
+Build
+====================
+
+ * For Metaware toolchain:
+```
+    $ cd <FreeRTOS_root>/FreeRTOS/Demo/ARC_NSIM_QEMU_GCC_MW
+    $ make
+```
+ * For arc_gnu toolchain:
+```
+    $ cd <FreeRTOS_root>/FreeRTOS/Demo/ARC_NSIM_QEMU_GCC_MW
+    $ make TOOLCHAIN=gnu
+```
+Debug
+====================
+
+This example outputs via console through Metaware or MDB debugger. 
+
+ * For Metaware toolchain:
+```
+    $ cd <FreeRTOS_root>/FreeRTOS/Demo/ARC_NSIM_QEMU_GCC_MW
+    $ make debug
+```
+Use nSIM simulator and debug in MetaWare IDE.
+
+ * For arc_gnu toolchain:
+
+nSIM simulator and QEMU are both available
+```
+    $ cd <FreeRTOS_root>/FreeRTOS/Demo/ARC_NSIM_QEMU_GCC_MW
+    $ make TOOLCHAIN=gnu nsim
+```
+This step is to launch nSIM simulator (to use QEMU, please replace nsim with qemu) and wait for the connection from gdb. Now open another terminal.
+```
+    $ cd <FreeRTOS_root>/FreeRTOS/Demo/ARC_NSIM_QEMU_GCC_MW
+    $ make TOOLCHAIN=gnu debug
+```
+Type c to continue and see the output.
+
+Sample Output
+=============
+
+If you are using the **nSIM**, one console msg wil be used to simulate the
+project, you will see the following output in the CMD window when the
+download is successful.
+```
+    TimerCounter=1  QueueCounter=4
+                                TimerCounter=2  QueueCounter=9
+                                                                TimerCounter=3  QueueCounter=14
```

Comment:
The original output is this...

Reply:
on Linux

Reply:
Is it launched via mdb or as a standalone nsim (nsimdrv)?

Reply:
nsimdrv and arc-elf32-gdb

Reply:
Hmm, may I ask you to check if we have same output without debugger on nSIM?

Reply:
still have that shifted log

Reply:
Such nSIM console works fine in case of linux / uboot / zephyr / etc. Is there a chance that we print something which we shouldn't? Or probably nSIM requre /r/n instead of /n?

Reply:
Thanks, add \r in sprintf() works fine right now. It's weird that qemu works fine without adding \r.